caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -120,28 +120,131 @@ def load(cls, model_path: str, *args, **kwargs) -> "EmbeddingModule":
 
         return cls.bootstrap(model_name_or_path=artifacts_path)
 
+    def _truncate_input_tokens(
+        self, truncate_input_tokens, texts: List[str]
+    ) -> List[str]:
+        """Truncate input tokens
+        Args:
+            truncate_input_tokens: int
+                Truncation length for input tokens.
+                If less than zero, this is disabled (returns texts without processing).
+                If zero or greater than the model's maximum, then this is a test
+                to see if truncation is needed. If needed, an exception is thrown.
+                Otherwise, we take this usable truncation limit to truncate the tokens and then
+                decode them to return truncated strings that can be used with this model.
+            texts: List[str]
+                Input texts to be checked and optionally truncated.
+        Returns:
+            List[str]: the texts after checking and/or truncating
+        """
+
+        # NOTE: When inference is called immediately after load (typical case with lazy loading),
+        # using the tokenizer right away here results in a warning like:
+        #     huggingface/tokenizers: The current process just got forked, after parallelism
+        #     has already been used. Disabling parallelism to avoid deadlocks...
+        #     To disable this warning, you can either:
+        #     - Avoid using `tokenizers` before the fork if possible
+        #     - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
+        # A warmup encode() call in load() will take care of this (the first option above).
+        # This here comment is in case we need to set TOKENIZERS_PARALLELISM in the future.
+
+        if truncate_input_tokens < 0:
+            return texts
+
+        max_tokens = self.model.max_seq_length
+
+        # Do truncation if given a usable truncation value, else test for need to truncation
+        if 0 < truncate_input_tokens <= max_tokens:
+            okay_to_truncate = True
+            max_length = truncate_input_tokens
+            ret = []  # will return truncated texts
+        else:
+            okay_to_truncate = False
+            max_length = max_tokens
+            ret = texts  # will not alter texts when truncation is not allowed
```

Comment:
In this case, we don't allow truncation but raise an exception which is returned.  So the texts are returned as-is only if we get through that.  Have to loop through to determine if the model would in fact do truncation then do the `elif was_truncated` ... `error.log_raise` (below).  

Unfortunately sentence-transformers doesn't just let me pass some parameter to behave this way :(. So truncation would happen with sentence-transformers, but the feature requested here is to raise an exception instead (to be consistent with other services/user experiences).

Since you mentioned "last error", I should point out that this raises an exception as soon as unwanted truncation is detected. There isn't any attempt to look for more errors right now - just fail fast.  The unit test is mostly checking that the exception is thrown when expected.

```
    model.run_embedding(text=ok)
    with pytest.raises(ValueError):
        model.run_embedding(text=too_long)
```

Reply:
added more tests

Reply:
I see. My main concern here was if more "work" was being done than necessary